include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1028,52 +1030,91 @@ struct __early_exit_find_or
 {
     _Pred __pred;
 
+    // operator() overload for __parallel_or_tag
+    template <typename _NDItemId, typename _IterSize, typename _WgSize, typename _FoundLocalState, typename _Compare,
+              typename... _Ranges>
+    void
+    operator()(const _NDItemId __item_id, const _IterSize __n_iter, const _WgSize __wg_size, _Compare __comp,
+               _FoundLocalState& __found_local, __parallel_or_tag, _Ranges&&... __rngs) const
+    {
+        const auto __n = oneapi::dpl::__ranges::__get_first_range_size(__rngs...);
+
+        std::size_t __shift = 16;
+        const std::size_t __local_idx = __item_id.get_local_id(0);
+        const std::size_t __group_idx = __item_id.get_group(0);
+
+        // each work_item processes N_ELEMENTS with step SHIFT
+        const std::size_t __leader = (__local_idx / __shift) * __shift;
+        const std::size_t __init_index =
+            __group_idx * __wg_size * __n_iter + __leader * __n_iter + __local_idx % __shift;
+
+        // if our "line" is out of work group size, reduce the line to the number of the rest elements
+        if (__wg_size - __leader < __shift)
+            __shift = __wg_size - __leader;
+        for (_IterSize __i = 0; __i < __n_iter; ++__i)
+        {
+            // Point #B1 - not required to have _ShiftedIdxType
+
+            _IterSize __current_iter = __i;
```

Comment:
Could you use the loop induction variable `__i` directly in the following lines instead of introducing this new variable `__current_iter`?

Reply:
Fixed